donations/views.py

```diff
@@ -77,43 +77,51 @@ def _save_donation(encoded_data, email, amount, currency, transaction_id, source
 @csrf_exempt
 def donation_complete_stripe(request):
     """
-    This view receives a stripe token of a validated credit card and requests for the actual donation.
-    If the donation is successfully done it stores it and send the email to the user.
+    This view is called from Stripe when a new donation is completed, here we create and
+    store the donation in the db.
     """
-    donation_received = False
-    donation_error = False
-    if request.method == 'POST':
-        form = DonateForm(request.POST, user=request.user)
+    if "HTTP_STRIPE_SIGNATURE" in request.META:
+        payload = request.body
+        sig_header = request.META['HTTP_STRIPE_SIGNATURE']
+        endpoint_secret = settings.STRIPE_WEBHOOK_SECRET
+        stripe.api_key = settings.STRIPE_PRIVATE_KEY
+        event = None
 
-        token = request.POST.get('stripeToken', False)
-        email = request.POST.get('stripeEmail', False)
+        try:
+            event = stripe.Webhook.construct_event(
+              payload, sig_header, endpoint_secret
+            )
+        except ValueError as e:
+            # Invalid payload
+            return HttpResponse(status=400)
```

Comment:
This is stripe making a request to the URL we provide right? So if this error happens this will be the response to stripe's request?

Reply:
If for some reason we return an error Stripe will retry later, but also someone could call this url, in this case, we also want to return an error.